### ISSUE TITLE
Adguard crashing after a weeks

### ISSUE THREAD

**Issue by mkono87** [mimugmail/opn-repo#95]
I have been dealing with this weird issue for months now and cant seem to pinpoint it as its somewhere rare and cant see anything in logs.

Out of the blue adguard will just stop running and I can either attempt to restart the service or I have to reboot the modem to get it running again. Do you know of a specific place I can check. Im thinking im either going to have to move adguard to another system or something if I cant get this sorted out. Any help would be appreciated.

**Comment by mimugmail** [comment]
Nothing in system log? So you Monitor RAM? Latest version?

**Comment by mkono87** [comment]
Not anything that sticks out as an issue. Nothing related to adguard either. Ram has been fine. I still have access to LAN and opnsense itself. I always keep adguard up to date from within the GUI. In terms of opnsense versions, I have had this issue over a few upgrades. So weird.

**Comment by ErwinJoubert** [comment]
Same here. Logs look clean as well.

Config is at `/usr/local/AdGuardHome/AdGuardHome.yaml`

**Comment by burntoc** [comment]
Still nothing, I guess?  I'm in the same boat - it's been happening for the year or so I've been using it and nothing is identifiable re: memory or logs.  I have to manually start it again - if I try restarting otherwise it stops running if it was and just stays stopped, but the manual start works for some reason.

**Comment by mkono87** [comment]
About a month ago, I reinstalled opnsense and its still happening from time to time.

**Comment by mimugmail** [comment]
Hard to troubleshoot without any logs or oom indicators. I'm running it at home without issues for months.

**Comment by burntoc** [comment]
How do we get the logs to provide more info than they normally do?  Nothing
obvious in the logs to date - still crashes at least monthly.

**Comment by mkono87** [comment]
@mimugmail  Are you able to tell us where we could find logs related to adguard? We have clean logs for what we can see.

**Comment by chrisf4lc0n** [comment]
@mkono87 I have experience the same problem. As a workaround I have created a Monit Service to monitor the AdGaurdHome with 5 seconds Polling Interval:
![Screenshot from 2022-04-29 16-46-31](https://user-images.githubusercontent.com/7551751/165979252-dc567eb2-94ef-4121-9e86-c7f064eb4180.png)

**Comment by burntoc** [comment]
> @mkono87 I have experience the same problem. As a workaround I have created a Monit Service to monitor the AdGuarddHome with 5 seconds Polling Interval: ![Screenshot from 2022-04-29 16-46-31](https://user-images.githubusercontent.com/7551751/165979252-dc567eb2-94ef-4121-9e86-c7f064eb4180.png) If it goes down the Monit will start it 10 seconds.

Thanks - I've been looking for something straightforward like this.  Can you ELI5 how I could set the poll time?  I have the normal Monit setting at 120, and I see if I click Advanced on this Service it shows Poll Time, but I thought it accepts values in seconds and if I type in a number it won't allow me to apply the changes.  Left it blank for now, but I'd like to increase it to 15 seconds.

**Comment by burntoc** [comment]
> @burntoc have you enabled the Monit?

Yep.  That's so odd.  As long as I leave Poll Time blank and click Apply it processes a fraction of a second and that notice disappears.  If I put any number in Poll Time, exactly as you've done, it will process very briefly as before, but the Apply notice stays there like it didn't fully take.  I can navigate away and back and it still shows the number.  Maybe it is just an interface bug.

**Comment by burntoc** [comment]
> @burntoc have you tried restarting the OPNsense box?

Just did.  Didn't help.  Reinstalled the package.  Didn't help.  Recreated the task - triple-checked to make sure it matches yours exactly, verified the path via ssh - no joy.  I just decided to leave that field blank for now and I changed the General Settings poll time to 15 and that change took okay.  I don't really have any other checks going on so hopefully that will work okay.

**Comment by bachya** [comment]
I've experienced this on OPNsense 22.1.10-amd64 twice in the last two weeks. With the AGH service down, I can't navigate to anything (LAN or WAN). I must wire up to the OPNsense box and manually restart the service. I did notice via the AGH UI that there was an update to 0.107.8, which I just applied; will see if that helps.

**Comment by aleksasiriski** [comment]
> ![Screenshot from 2022-04-29 17-14-29](https://user-images.githubusercontent.com/7551751/165983432-4a5d6a92-e7bc-493d-886a-c5723d598467.png)

Poll Time for 5 seconds needs: **5 CYCLES** instead of just *5*

**Comment by hlarsen** [comment]
For any others troubleshooting this there's a `log_file` option in `AdGuardHome.yaml` - I've only has this happen twice, so hopefully if it happens again there will be some further info in the AdGuard log.

**Comment by MadWalnut** [comment]
Just had this happen twice within a few days. Has anyone found any details why it occurs? Will be trying the monit workaround and the `log_file` option next.
